include/connection.h

```diff
@@ -50,10 +49,27 @@ typedef struct NodeConnectionEntry
 } NodeConnectionEntry;
 
 
+
+typedef struct {
+	ShardId shardId;
+	int nReplicas;
+	bool* status;
```

Comment:
My intention was to leave semantic of "status" field for users of DoForAllShards function - for callbacks passed in this function. So there are two options:
1. Make it boolean with semantic: ok/error
2. Make it integer and let callback interpret it any way it like. But in this case ShardAction should also return integer, not boolean. But then return code can not be interpreted by DoForAllShards to stop iteration as soon as possible when error is encountered and return ID of failed shard.

This is why I have chosen option 1.